zerver/actions/message_edit.py

```diff
@@ -1340,6 +1347,49 @@ def check_update_message(
             mentioned_group_ids = list(rendering_result.mentions_user_group_ids)
             check_user_group_mention_allowed(user_profile, mentioned_group_ids)
```

Comment:
I think would be good if we can just keep this code to check content edit permissions in the `check_update_message` only, having this function returns `mentioned_group_ids`, etc. and other multiple fields makes it hard to read.

Though we do not allow it in the UI, but I imagine there can be a API request for editing content and also resolving topic, so we should make sure handle it correctly if we allow such requests.